src/Simulation/qdk_sim_rs/src/linalg/expm.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::ops::Mul;
+
+use cauchy::Scalar;
+
+#[cfg(feature = "pade")]
+use ndarray::{ArrayBase, Data, Ix2, OwnedRepr, RawData, ScalarOperand};
+
+use crate::linalg::decompositions::{EigenvalueDecomposition, ExplicitEigenvalueDecomposition};
+
+#[cfg(feature = "pade")]
+use crate::{
+    error::QdkSimError,
+    linalg::{Identity, Inv, MatrixPower},
+    math::approximate_factorial,
+};
+
+// NB: We use the notation of
+//     https://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Moler03.pdf.
+//
+//     In particular:
+//
+//         e^{A} ≈ R_pq(A) ≔ [D_pq(A)]⁻¹ N_pq(A)
+//
+//     where
+//
+//         N_pq(A) ≔ Σ_j=0^p [(p + q - j)! p!] / [(p + q)! j! (p - j)!] A^j
+//
+//     and
+//
+//         D_pq(A) ≔ Σ_j=0^q [(p + q - j)! p!] / [(p + q)! j! (q - j)!] (-A)^j.
```

Comment:
Consider removing empty lines:
```suggestion
// NB: We use the notation of
//     https://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Moler03.pdf.
//     In particular:
//         e^{A} ≈ R_pq(A) ≔ [D_pq(A)]⁻¹ N_pq(A)
//     where
//         N_pq(A) ≔ Σ_j=0^p [(p + q - j)! p!] / [(p + q)! j! (p - j)!] A^j
//     and
//         D_pq(A) ≔ Σ_j=0^q [(p + q - j)! p!] / [(p + q)! j! (q - j)!] (-A)^j.
```

Reply:
I personally tend to prefer blank lines around formula, as that's what I'm used to from technical publishing in mathematics, but happy to make the change if you feel strongly about it.

Reply:
My opinion is not strong, feel free to leave as is.